initializing backend repository

umutoksuz opened the pull request:
# Initializing backend repository

- [x] New feature (non-breaking change which adds functionality)


# Description

This pull request initializes our Django project subrepo for backend.

I have installed following packages:
- Django (our core framework)
- Django rest framework (provides additional tools for api development)
- psycopg2 (enables us to work with postgresql)
- pylint-django (needed for queries)

getflix folder keeps projects initalization, setting kind of files. API folder contains out functionality.
For database connection I will provide you a 'secrets.json' file. After secret file is provided you can run project by using 
'python manage.py runserver' commmand.


## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have checked my code and corrected any misspellings
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

umutoksuz committed:
Initialize backend repo

MehdiSaffar committed:
remove extra 'getflix' folder, .gitignore

MehdiSaffar commented:
I did some tiny changes to the folder structure (there was an extra getflix folder).
I added a local .gitignore there and moved the secrets.json entry in the global .gitignore to this local one as well as add python/django specific gitignore.

Please do the following:
1. Add a README.md explaining how to setup the venv, install packages, make it work. Please take a look at mine or Asli's for an idea.
2. Add a .dockerfile so that people can use it during development and to be ready for production environment.

Questions:
1. Is that db.sqlite3 temporary? Should it be .gitignore'ed?

When about to accept this pull request, please do NOT merge. Instead, click on the arrow and select Squash and merge. When editing the commit message, tidy it up (look at init frontend or init android commit in master) then finish.

![image](https://user-images.githubusercontent.com/13124263/98588908-82a68500-22dd-11eb-8551-d970bd2fbfc1.png)


Thanks

umutoksuz commented:
> I did some tiny changes to the folder structure (there was an extra getflix folder).
> I added a local .gitignore there and moved the secrets.json entry in the global .gitignore to this local one as well as add python/django specific gitignore.
> 
> Please do the following:
> 
> 1. Add a README.md explaining how to setup the venv, install packages, make it work. Please take a look at mine or Asli's for an idea.
> 2. Add a .dockerfile so that people can use it during development and to be ready for production environment.
> 
> Questions:
> 
> 1. Is that db.sqlite3 temporary? Should it be .gitignore'ed?
> 
> When about to accept this pull request, please do NOT merge. Instead, click on the arrow and select Squash and merge. When editing the commit message, tidy it up (look at init frontend or init android commit in master) then finish.
> 
> ![image](https://user-images.githubusercontent.com/13124263/98588908-82a68500-22dd-11eb-8551-d970bd2fbfc1.png)
> 
> Thanks

Thank you for your review.

Folder structure was created by Django it duplicates your project folder first contains the manage.py and folder for other .py files.

It would be better if you tell us your feedback rather than doing changes directly in backend branches. According to your feedback we can discuss the situation and apply changes accordingly.
